libcylinder/src/signing/secp256k1.rs

```diff
@@ -53,7 +52,9 @@ pub struct Secp256k1PrivateKey {
 
 impl Secp256k1PrivateKey {
     pub fn from_hex(s: &str) -> Result<Self, Error> {
-        hex_str_to_bytes(s).map(|key_bytes| Secp256k1PrivateKey { private: key_bytes })
+        hex_str_to_bytes(s)
+            .map(|key_bytes| Secp256k1PrivateKey { private: key_bytes })
+            .map_err(|err| Error::SigningError(Box::new(err)))
```

Comment:
This has changed the error type.  The previous error returned here was a `ParseError`, it also doesn't seem like you're using the enum created in the error module.

Reply:
Good catch, switched it to `ParseError`. This method is removed in another commit.